distribution/lib/Standard/Base/0.0.0-dev/src/Polyglot/Proxy_Polyglot_Array.enso

```diff
@@ -0,0 +1,27 @@
+from Standard.Base import Polyglot, Array
+
+## Advanced
+
+    Wrapper for Polyglot Arrays
+type Proxy_Polyglot_Array
+
+    type Proxy_Polyglot_Array arr
+
+    ## Returns the number of elements stored in this Polyglot Array.
+
+    length : Number
+    length self =
+        Polyglot.get_array_size self.arr
+
+    ## Gets an element from this Polyglot Array at a specified index (0-based).
+
+    at : Number -> Any
+    at self index =
+        Polyglot.read_array_element self.arr index
+
+    to_array : Array Any
+    to_array self =
+        len = self.length
+        a = Array.new len
+        0.up_to len . each ix-> a.set_at ix (self.at ix)
```

Comment:
This seems very inefficient.

Any reason why you cannot replace it with `Array.copy self.arr 0 a 0 len`?

Reply:
Also I'm a bit worried about the fact that we would always copy here. Cannot we somehow wrap the polyglot array while avoiding copying? At least for cases where we can guarantee immutability as we were discussing.

Reply:
It is pretty much as inefficient as it was before, really. I would defer optimizations until we actually decide to rewrite Array, and we seem to be coming to that point very soon :)

Reply:
What are you comparing to? Before we didn't have a way to convert a polyglot array to Enso, `to_array` would just return it - which was a zero-cost  operation.

Reply:
to (now gone) `from_array`

Reply:
Ah, ok, right.

I was thinking of the `Vector.Vector polyglot_array` approach which we used most of the time.